02_knight_and_princess/tevfik_aksoy/main.cpp

```diff
@@ -0,0 +1,147 @@
+#include <array>
+#include <fstream>
+#include <iostream>
+#include <set>
+#include <string>
+#include <tuple>
+#include <vector>
+
+using std::array;
+using std::endl;
+using std::get;
+using std::to_string;
+using std::vector;
+
+using VertexId = int;
+using EdgeId = int;
+
+// Tuple of <int,int> for realization of vector of vectors of <int,int>
+using dest_id_edge_id = std::tuple<int, int>;
+
+struct Vertex {
+ public:
+  VertexId id = 0;
+
+  explicit Vertex(VertexId _id = 0) : id(_id) {}

```

Comment:
Есть ли смысл создавать `Vertex` с `id = 0` по-умолчанию? А если мы создадим 10 таких вершин, получится что в графе есть 10 вершин с одинаковым id? Наверное, лучше будет не указывать дефолтное значение?
```suggestion
  explicit Vertex(VertexId _id) : id(_id) {}
```

Reply:
По значению мы должны передавать только фундаментальные типа, но так как в данном случае конкретный тип скрыт от нас за `VertexId`, мы не знаем, фундаментальный это тип или нет, в таком случае лучше пользоваться стандартной практикой передачи по `const` ссылке

Reply:
Здравствуйте, 
![image](https://user-images.githubusercontent.com/24300906/136514520-80560b26-a7f7-485f-8ec5-93d704f28de9.png)
получаю такую ошибку если не создаю с значением по умолчанию, этот конструктор работал еще как дефольтный. Тут мне кажется либо надо создать дефольтный конструктор что и сделает то же самое с этим, либо вернуть array на vector, да?